src/ProtoRoute/Message.hs

```diff
@@ -1,26 +1,35 @@
 module ProtoRoute.Message
-    ( constructProtoMessage
-    , MessageName
-    , FieldName
-    , FieldValue
-    ) where
+    ( MsgName (..)
+    , FieldName (..)
+    , FVType (..)
+    , Msg (..)
+    , FieldValue (..)
+    , constructProtoMsg) where
 
-import Data.ProtoLens
-import Data.Text (pack)
-import ProtoExports
+import Data.Text (pack, Text)
 
+newtype MsgName   = MN Text deriving (Show)
+newtype FieldName = FN Text deriving (Show)
 
-newtype MessageName = MessageName { unMessageName :: Text }
-newtype FieldName   = FieldName   { unFieldName   :: Text }
-data    FieldValue  = FV [Int] | FV [Text] | FV [FieldValue]
+data FieldValue a = Req a | Opt (Maybe a) | Rep [a] deriving (Show)
+data FVType       = FieldText (FieldValue Text) | FieldInt (FieldValue Int) |
+                    FieldMsg (FieldValue Msg) deriving (Show)
+-- Typesafe representation of a protobuf mesage
+data Msg = Msg
+    { messageName :: MsgName
+    , messageFields :: [(FieldName, FVType)]
+    } deriving (Show)
 
-constructProtoMessage :: MessageName -> [(FieldName, FieldValue)] -> Text
-constructProtoMessage msgName namesVals = msgName ++ pack " {"
+constructProtoMsg :: MsgName -> [(FieldName, FVType)] -> Text
+constructProtoMsg msgName namesVals =
+    pack (show msgName ++ " {" ++ addRest msgName namesVals ++ "} ")
```

Comment:
You're calling `execStream` function from `ghcid` package. And this functions takes `Text`. So it doesn't actually makes much sense to use `show` to convert to `Text` and then call `unpack` to convert back to `String`. You can work directly with `String` to make life easier.